plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/RocksDBKeyValueStorageFactory.java

```diff
@@ -177,19 +181,35 @@ public KeyValueStorage create(
                 segments.stream()
                     .filter(segmentId -> segmentId.includeInDatabaseVersion(databaseVersion))
                     .collect(Collectors.toList());
-
-            segmentedStorage =
-                new RocksDBColumnarKeyValueStorage(
-                    rocksDBConfiguration,
-                    segmentsForVersion,
-                    ignorableSegments,
-                    metricsSystem,
-                    rocksDBMetricsFactory);
+            if (isForestStorageFormat) {
+              LOG.info("FOREST mode detected, using pessimistic DB.");
+              segmentedStorage =
+                  new PessimisticRocksDBColumnarKeyValueStorage(
+                      rocksDBConfiguration,
+                      segmentsForVersion,
+                      ignorableSegments,
+                      metricsSystem,
+                      rocksDBMetricsFactory);
+            } else {
+              LOG.info("Using Optimistic DB.");
+              segmentedStorage =
+                  new OptimisticRocksDBColumnarKeyValueStorage(
+                      rocksDBConfiguration,
+                      segmentsForVersion,
+                      ignorableSegments,
+                      metricsSystem,
+                      rocksDBMetricsFactory);
+            }
           }
           final RocksDbSegmentIdentifier rocksSegment =
               segmentedStorage.getSegmentIdentifierByName(segment);
-          return new SegmentedKeyValueStorageAdapter<>(
-              segment, segmentedStorage, () -> segmentedStorage.takeSnapshot(rocksSegment));
+
+          if (isForestStorageFormat) {
+            return new SegmentedKeyValueStorageAdapter<>(segment, segmentedStorage);
+          } else {
+            return new SegmentedKeyValueStorageAdapter<>(
```

Comment:
I am not a fan of our adapter pattern for rocksdb, HOWEVER it would be clarifying if forest had its own adapter that did not implement SnappableKeyValueStorage.  As it is, Pessimistic db would report as being Snappable.  Forest code never takes a snapshot, so it is mostly academic, but could lead to problems in the future.

Reply:
Done.